test/test-migrate-export.sh

```diff
@@ -0,0 +1,432 @@
+  assert_local_object "$md_feature_oid" "30"
+
+  master="$(git rev-parse refs/heads/master)"
+  feature="$(git rev-parse refs/heads/my-feature)"
+
+  master_attrs="$(git cat-file -p "$master:.gitattributes")"
+  feature_attrs="$(git cat-file -p "$feature:.gitattributes")"
+
+  echo "$master_attrs" | grep -q "*.md text !filter !merge !diff"
+  echo "$master_attrs" | grep -q "*.txt text !filter !merge !diff"
+
+  [ ! $(echo "$feature_attrs" | grep -q "*.md text !filter !merge !diff") ]
+  [ ! $(echo "$feature_attrs" | grep -q "*.txt text !filter !merge !diff") ]
+)
+end_test
+
+begin_test "migrate export (with remote)"
+(
+  set -e
+
+  setup_single_remote_branch_tracked
+
+  git push origin master
+
+  md_oid="$(calc_oid "$(cat a.md)")"
+  txt_oid="$(calc_oid "$(cat a.txt)")"
+
+  assert_pointer "refs/heads/master" "a.md" "$md_oid" "50"
+  assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "30"
+
+  assert_pointer "refs/remotes/origin/master" "a.md" "$md_oid" "50"
+  assert_pointer "refs/remotes/origin/master" "a.txt" "$txt_oid" "30"
+
+  # Flush the cache to ensure all objects have to be downloaded
+  rm -rf .git/lfs/objects
+
+  git lfs migrate export --everything --include="*.md, *.txt"
+
+  [ ! $(assert_pointer "refs/heads/master" "a.md" "$md_oid" "50") ]
+  [ ! $(assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "30") ]
+
+  [ ! $(assert_pointer "refs/remotes/origin/master" "a.md" "$md_oid" "50") ]
+  [ ! $(assert_pointer "refs/remotes/origin/master" "a.txt" "$txt_oid" "30") ]
+
+  # All pointers have been exported, so all objects should be pruned
+  refute_local_object "$md_oid" "50"
+  refute_local_object "$txt_oid" "30"
+
+  master="$(git rev-parse refs/heads/master)"
+  master_attrs="$(git cat-file -p "$master:.gitattributes")"
+
+  echo "$master_attrs" | grep -q "*.md text !filter !merge !diff"
+  echo "$master_attrs" | grep -q "*.txt text !filter !merge !diff"
+)
+end_test
+
+begin_test "migrate export (include/exclude args)"
+(
+  set -e
+
+  setup_single_local_branch_tracked
+
+  md_oid="$(calc_oid "$(cat a.md)")"
+  txt_oid="$(calc_oid "$(cat a.txt)")"
+
+  assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "120"
+  assert_pointer "refs/heads/master" "a.md" "$md_oid" "140"
+
+  git lfs migrate export --include="*" --exclude="a.md"
+
+  [ ! $(assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "120") ]
+  assert_pointer "refs/heads/master" "a.md" "$md_oid" "140"
+
+  refute_local_object "$txt_oid" "120"
+  assert_local_object "$md_oid" "140"
+
+  master="$(git rev-parse refs/heads/master)"
+
+  master_attrs="$(git cat-file -p "$master:.gitattributes")"
+
+  echo "$master_attrs" | grep -q "* text !filter !merge !diff"
+  echo "$master_attrs" | grep -q "a.md filter=lfs diff=lfs merge=lfs"
+)
+end_test
+
+begin_test "migrate export (bare repository)"
+(
+  set -e
+
+  setup_single_remote_branch_tracked
+  git push origin master
+  make_bare
+
+  git lfs migrate export --everything --include="*"
```

Comment:
Should there be any assertions following this? I think that the corresponding `migrate import` test is blank because it was simply fixing a crash, but we should check that the export works correctly here.